2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+    - Zarhus OS.
+- Outro and sources.
+
+???
+
+- Time for this slide: 30s
+- Idea/goal of this slide: Present entire presentation in a short way
+- What to say:
+    - TODO
+- Notes:
+    - TODO
+
+---
+
+# What is Trusted Execution Environment?
+
+.center[ <img src="/img/TEE_High_Lvl_Overview.svg" height="250px"> ]
+
+<br>
+
+_"A trusted execution environment (TEE) means a secure area which can guarantee
+the confidentiality and integrity of the code and data inside of this area._
+
+???
+
+- "from general information to specifics"
+- There's no agreed on firm definition and a lot of sources define their own
+- Definition taken from _"Building Secure Firmware, Armoring the Foundation of
+the Platform"_
+  + Will be listed in resources
+  + **Usually** refers to the possibility of TEE being only isolated via
+software which is not the case in some definitions.
+- Can be isolated via software
+- The term TEE was coined somewhere in 2000s
+- Used to be "secure area of the main processor" but the definition has
+broadened
+  + Can even be a separate device, more on that later
+- A device can have multiple TEEs nowadays
+- Used implementation is mostly dependant on underlying CPU architecture
+
+---
+
+# Common Ground in TEE definition
+
+### What it ensures:
+
+- **Data integrity**: preventing unauthorized entities from altering data when
+data is being processed.
+
+- **Data confidentiality**: Guarantees the authenticity of the executed code.
+
+- **Code integrity**: The confidentiality of its code, data and runtime states
+stored on a persistent memory.
+
+???
+
+- This is the part that is universally agreed on
+- This exact definition is from Confidential Computing Consortium
+  - A Linux Foundation Project which aims to accelerate the adoption of Trusted
+  Execution Environment (TEE) technologies and standards
+  - With Confidential Computing being the protection of data in use by
+  performing computation in a hardware-based, attested Trusted Execution
+  Environment.
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**CPU based TEE**
+
+<img src="/img/TEE_CPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Arm Trustzone
+- Intel SGX (Software Guard Extensions)
+- AMD SEV (Secure Encrypted Virtualization)
+
+]
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**Coprocessor Based TEE (One SoC)**
+
+<img src="/img/TEE_CoCPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Integrated TPM (Trusted Platform Module)
+- Apple Secure Enclave
+
+]
+
+???
+
+- ARM Cortex-M based Secure Subsystem
+- Shared resources
+  - Power, memory, caches, buses, clock signals
+  - Larger vector of attack
+- Hardware/firmware vulnerabilities could be exploited
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**Coprocessor Based TEE (External)**
+
+<img src="/img/TEE_SepCPU_based.svg" width="240px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Discrete TPM (Trusted Platform Module)
```

Comment:
Is TPM a TEE? It is rather part of it.

Reply:
Same response as [above](https://github.com/3mdeb/conferences/pull/36#discussion_r1763925746)

Reply:
I see, not critical for now then.